spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -101,15 +118,50 @@ private void validateAndSetFilterParts() {
 
 	@Override
 	public Object execute(Object[] parameters) {
+		Class<?> returnedObjectType = getQueryMethod().getReturnedObjectType();
+		Object result;
+		if (isSliceQuery() || isPageQuery()) {
+			List<?> resultEntries = (List) execute(parameters, returnedObjectType, List.class, false);
+			Long totalCount = (Long) execute(parameters, Long.class, null, true);
```

Comment:
I understand that there isn't another way to get the count other than querying again for it.
This work is definitely worth putting in since it's done, but I question the original premise that because the slice/param feature is useful in old-school SQL jpa spring data that it also makes sense here in Datastore.

Reply:
Have you looked into [datastore statistics](https://cloud.google.com/datastore/docs/concepts/stats)?
At the very least we should clearly document this inefficiency in the refdoc.

Reply:
noting this just so it's here for reference, but after looking into the stats tables it turns out they are not updated in real time (more like at a few points per day), so they cannot be used for our purposes.